src/App.tsx

```diff
@@ -14,210 +34,132 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const getUserTodos = useCallback(async () => {
+    if (user) {
+      try {
+        setIsError(false);
+
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setErrorText('Unable to connect to server');
+        setIsError(true);
+      }
+    }
+  }, []);
+
+  const getVisibleGoods = useCallback((allTodos: Todo[]) => {
+    return allTodos.filter(todo => {
+      switch (filterStatus) {
+        case StatusToFilter.Active:
+          return !todo.completed;
+
+        case StatusToFilter.Completed:
+          return todo.completed;
+
+        case StatusToFilter.All:
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterStatus]);
+
+  const addNewTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    try {
+      setIsError(false);
+      setIsAdding(true);
+
+      const newTodo = {
+        ...todoData,
+        id: Math.max(...todos.map(todo => todo.id)) + 1,
+      };
+
+      await addTodo(newTodo);
+
+      await getUserTodos();
+
+      setIsAdding(false);
+    } catch (error) {
+      setErrorText('Unable to add Todo');
+      setIsError(true);
+    }
+  }, []);
+
+  const deleteCurrentTodo = useCallback(async (todoId: number) => {
+    try {
+      setIsError(false);
+
+      setTodoIdsToDelete(prevTodoIds => [...prevTodoIds, todoId]);
+
+      await deleteTodo(todoId);
+
+      await getUserTodos();
+
+      setTodoIdsToDelete([0]);
+    } catch (error) {
+      setErrorText('Unable to delete Todo');
+      setIsError(true);
+    }
+  }, []);
+
+  const clearAllCompletedTodos = useCallback(async () => {
+    setIsError(false);
+
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setTodoIdsToDelete(prevTodoIds => (
+      [...prevTodoIds, ...completedTodos.map(todo => todo.id)]));
+
+    await Promise.all(completedTodos.map(async (todo) => {
+      await deleteCurrentTodo(todo.id);
```

Comment:
a liitle bit completed for me. I guess, you could make it without async function because you have **deleteCurrentTodo** and it is already async

Reply:
maybe you`re right, just thinked that it depends on deleting all todos in one time

Reply:
maybe it is, you can check it)